### PR TITLE
test(e2e): add s3 addon e2e test

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR:
1. add s3 e2e test (similar to #2147)
2. remove retain policy for ddb and s3 templates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .release/buildspec_e2e.yml [hunk 1]
```diff
@@ -19,6 +19,15 @@ batch:
         variables:
           TEST_SUITE: addons
           APP_REGION: us-west-2
+    - identifier: s3
```
Comment (efekarakus) [inline .release/buildspec_e2e.yml]: Ooh what do you think of moving all this code into the existing `addons` test suite?  
(Launch both an Aurora DB and S3 bucket, then test that the API works)

In the past we ran into issues when adding too many e2e tests within the same region, VPC limits, polling limits... so we try to group related functionality under the same test suite.

**Review by efekarakus — COMMENTED** [review]
The code itself looks great! Just a request to move it into the existing test suite 🙏

**Review by efekarakus — APPROVED** [review]

File: templates/addons/ddb/cf.yml [hunk 2]
```diff
@@ -13,7 +13,6 @@ Resources:
     Metadata:
       'aws:copilot:description': 'An Amazon DynamoDB table for {{.Name}}'
     Type: AWS::DynamoDB::Table
-    DeletionPolicy: Retain
```
Comment (iamhopaul123) [inline templates/addons/ddb/cf.yml:16]: If we are not using `storage init` to create an addons template. How do we make sure the template generated by `storage init` actually works 🤔

Comment (iamhopaul123) [reply templates/addons/ddb/cf.yml:16]: oh nvm. i thought this is in e2e tests. But why do we not want to retain ddb and s3 anymore?

Comment (Lou1415926) [reply templates/addons/ddb/cf.yml:16]: discussed offline!

**Review by bvtujo — APPROVED** [review]
Looks like a great test!

**Pull request merged**